### ISSUE TITLE
Support for flexible Policy enforcement during assignment

### ISSUE THREAD

**Issue by bhummerstone** [terraform-providers/terraform-provider-azurerm#5699]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

When assigning a policy in the Portal, you can enable/disable the Enforcement Mode; this is useful for using the same Policy definition (with a built-in Deny) but disabling enforcement in Non-Production environments.

### New or Affected Resource(s)

* azurerm_policy_assignment

### Potential Terraform Configuration

```hcl
resource "azurerm_policy_assignment" "example" {
  name                 = "example-policy-assignment"
  scope                = "${azurerm_resource_group.example.id}"
  policy_definition_id = "${azurerm_policy_definition.example.id}"
  description          = "Policy Assignment created via an Acceptance Test"
  display_name         = "My Example Policy Assignment"
  enforcement_mode = "disabled"

  parameters = <<PARAMETERS
{
  "allowedLocations": {
    "value": [ "West Europe" ]
  }
}
PARAMETERS
}
```

### References

None

**Comment by ccampo** [comment]
I have created a pull request that addresses this issue. Feedback welcome

https://github.com/terraform-providers/terraform-provider-azurerm/pull/7331

**Comment by bhummerstone** [comment]
Perfect! Thank you :)

**Comment by ArcturusZhang** [comment]
Should be resolved in #7331 , closing....
